cmake/CgvFindVersion.cmake

```
@@ -69,104 +69,160 @@ if(CMAKE_SCRIPT_MODE_FILE)
   cmake_minimum_required(VERSION 3.8)
 endif()
 
+#-----------------------------------------------------------------------------#
+
+function(_cgv_store_version string suffix hash)
+  set(_CACHED_VERSION "${string}" "${suffix}" "${hash}")
+  # Note: extra 'unset' is necessary if using CMake presets with
+  # ${CGV_PROJECT}_GIT_DESCRIBE=""
+  unset(${CGV_CACHE_VAR} CACHE)
+  set(${CGV_CACHE_VAR} "${_CACHED_VERSION}" CACHE INTERNAL
+    "Version string and hash for ${CGV_PROJECT}")
+endfunction()
+
+#-----------------------------------------------------------------------------#
+
+function(_cgv_try_archive_md)
+  # Get a possible Git version generated using git-archive (see the
+  # .gitattributes file)
+  set(_ARCHIVE_DESCR "$Format:%$")
+  set(_ARCHIVE_TAG "$Format:%D$")
+  set(_ARCHIVE_HASH "$Format:%h$")
+  if(_ARCHIVE_HASH MATCHES "Format:%h")
+    # Not a git archive
+    return()
+  endif()
+
+  string(REGEX MATCH "tag: *${CGV_TAG_REGEX}" _MATCH "${_ARCHIVE_TAG}")
+  if(_MATCH)
+    _cgv_store_version("${CMAKE_MATCH_1}" "${CMAKE_MATCH_2}" "")
+  else()
+    message(WARNING "Could not match a version tag for "
+      "git description '${_ARCHIVE_TAG}': perhaps this archive was not "
+      "exported from a tagged commit?")
+    string(REGEX MATCH " *([0-9a-f]+)" _MATCH "${_ARCHIVE_HASH}")
+    if(_MATCH)
+      _cgv_store_version("" "" "${CMAKE_MATCH_1}")
+    endif()
+  endif()
+endfunction()
+
+#-----------------------------------------------------------------------------#
+
+function(_cgv_try_git_describe)
+  # First time calling "git describe"
+  if(NOT Git_FOUND)
+    find_package(Git QUIET)
+    if(NOT Git_FOUND)
+      message(WARNING "Could not find Git")
```

Should we describe the consequence? 
```suggestion
      message(WARNING "Could not find Git.  The version number will not be representative of the tags.")
```

---

Good though .Since there's multiple failure modes this way, and since the warning may be lost among the initial cmake output, what I've done is to add an install-time warning message about the missing version that shows up as the last thing users see.

---

I would add to the final warning a (list of) hint(s) of what kind of error might have cause this (eg. `missing git, etc..`)